compositor_common/src/error.rs

```diff
@@ -44,6 +44,58 @@ impl Display for UnusedNodesError {
     }
 }
 
+#[derive(Debug, thiserror::Error, PartialEq, Eq)]
+pub enum ConstraintsValidationError {
+    #[error("Specification of node \"{1}\" doesn't satisfy node input pads count constraint.")]
+    InvalidInputsPads(#[source] InputsCountValidationError, NodeId),
+}
+
+#[derive(Debug, thiserror::Error, PartialEq, Eq)]
+pub enum InputsCountValidationError {
+    #[error(
+        "Nodes that use transformation \"{}\" doesn't expect input pads, but {} input pads were specified.",
+        transformation_name,
+        defined_input_pads_count
+    )]
+    NoInputsRequired {
+        transformation_name: Arc<str>,
+        defined_input_pads_count: u32,
+    },
+    #[error(
+        "Nodes that use transformation \"{}\" expect single input pad, but {} input pads were specified.",
+        transformation_name,
+        defined_input_pads_count
+    )]
+    ExactlyOneInputRequired {
+        transformation_name: Arc<str>,
+        defined_input_pads_count: u32,
+    },
+    #[error(
+        "Nodes that use transformation \"{}\" expect {} input pads, but {} input pads were specified.",
+        transformation_name,
+        expected_input_pads_count,
+        defined_input_pads_count
+    )]
+    ExactNumberOfInputsRequired {
+        transformation_name: Arc<str>,
+        expected_input_pads_count: u32,
+        defined_input_pads_count: u32,
+    },
+    #[error(
+        "Nodes that use transformation \"{}\" expect at least {} and at most {} input pads, but {} were specified", 
```

Comment:
There should be different message for case where you need more than x and when you need less than x.